### PR TITLE
[QC-602] Batch processing in QC

### PR DESCRIPTION
**Pull request by knopers8**
This allows to process partial data by QC Tasks and dump/merge results into a file.
Then, the merged results can be picked up by the last parts of the QC chain (Checks, Aggregators, storage).

### CODE REVIEW

**Comment by knopers8** [comment]
@Barthelemy This is essentially working and ready for a review, I only have to write tests now.
I am especially not sure about the naming, so any other suggestions for "local batch", "remote batch", "RootFileSink" and "RootFileSource" will be very welcome.

**Comment by Barthelemy** [comment]
I'll review it today.

**Comment by knopers8** [comment]
I am not sure it is related to tuning the test parameters https://github.com/AliceO2Group/QualityControl/pull/777
but three tests now failed in o2-dataflow:
```
## sw/BUILD/QualityControl-latest/log
--
 5/48 Test #38: testFactory .............................***Timeout  20.97 sec
--
 9/48 Test #16: testCcdbDatabase ........................***Timeout  30.37 sec
--
36/48 Test #32: multinode_test ..........................***Failed   42.98 sec
--
94% tests passed, 3 tests failed out of 48
```

**Comment by Barthelemy** [comment]
Weird. I did not touch `testCheckWorkflow` which is the only one I see actually failing. 

```
The following tests FAILED:
	 25 - testCheckWorkflow (Timeout)
Errors while running CTest
```

I will also increase the timeout for `testCheckWorkflow`

**Review by Barthelemy — COMMENTED** [review]
Good, the naming is ok, I don't have a better proposal. 

The infrastructure generator is what it is ... :)

I tried to run the test but I get: 
> Error, monitor object of the QC Task could not be found.

**Comment by knopers8** [comment]
The test could fail this way if you provide it with wrong UNIQUE_ID env var. It passes CI now, so I doubt this is a real problem.

File: Framework/script/o2-qc-batch-test.sh [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  trap "
+    cleanup
+    trap - $sig EXIT
+    kill -s $sig "'"$$"' "$sig"
+done
+#trap cleanup EXIT
+
+function delete_data() {
+  curl -i -L ccdb-test.cern.ch:8080/truncate/qc/TST/MO/BatchTestTask${UNIQUE_ID}*
+  curl -i -L ccdb-test.cern.ch:8080/truncate/qc/TST/QO/BatchTestCheck${UNIQUE_ID}*
```
[the remark this answers is not in the document]
Comment (knopers8) [reply Framework/script/o2-qc-batch-test.sh:26]: Good point, I will add it (and also for multinode test) in another PR.

**Review by Barthelemy — APPROVED** [review]

**Pull request merged**